lib/src/types.ts

```diff
@@ -128,27 +146,32 @@ export interface ObjectPropertiesType {
 export interface ObjectType {
   kind: TypeKind.OBJECT;
   properties: Array<ObjectPropertiesType>;
+  schemaProps?: SchemaProp[];
 }
 
 export interface ArrayType {
   kind: TypeKind.ARRAY;
   elementType: Type;
+  schemaProps?: SchemaProp[];
 }
 
 export interface UnionType {
   kind: TypeKind.UNION;
   types: Type[];
   discriminator?: string;
+  schemaProps?: SchemaProp[];
 }
 
 export interface IntersectionType {
   kind: TypeKind.INTERSECTION;
   types: Type[];
+  schemaProps?: SchemaProp[];
 }
 
 export interface ReferenceType {
   kind: TypeKind.REFERENCE;
   name: string;
+  schemaProps?: SchemaProp[];
```

Comment:
Do any of these non-primitive types require `schemaProps`?

Reply:
Fixed for ReferenceType & NullType